### PR TITLE
Document PostgreSQL type mapping

### LINKED ISSUE #5131: Add type mapping documentation for PostgreSQL connector

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Document PostgreSQL type mapping

## Related issues, pull requests, and links

* Fixes #5131

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

#

Screen Shot 2022-03-09 at 15 06 46

### CODE REVIEW

**Comment by mosabua** [comment]
I cant comment on technical accuracy but typically there are different mappings for read vs create table style operations. 

https://trino.io/docs/current/connector/oracle.html#type-mapping

And we probably should also import https://trino.io/docs/current/connector/redshift.html#id1

**Comment by mosabua** [comment]
The best wording that clarifies the two mappings can be found in https://docs.starburst.io/latest/connector/starburst-synapse.html#type-mapping .. we have plans to make all connectors in Trino consistent along those lines but have not gotten around to it. Maybe we can use this PR as model to establish a good baseline wording for this in Trino.

**Comment by ebyhr** [comment]
> And we probably should also import https://trino.io/docs/current/connector/redshift.html#id1

The "General configuration properties" already exists in the PostgreSQL documentation.

File: docs/src/main/sphinx/connector/postgresql.rst [hunk 1]
```diff
@@ -93,6 +93,34 @@ catalog named ``sales`` using the configured connector.
...
 
+The data type mappings are as follows:
+
+=================== ================================ =======================================================================
+PostgreSQL          Trino                            Notes
+=================== ================================ =======================================================================
+``BIT``             ``BOOLEAN``
+``SMALLINT``        ``SMALLINT``
+``INTEGER``         ``INTEGER``
+``BIGINT``          ``BIGINT``
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/postgresql.rst:104]: This reminds that we should probably also test smallserial, serial and bigserial types too.

File: docs/src/main/sphinx/connector/postgresql.rst [hunk 2]
```diff
@@ -93,6 +93,34 @@ catalog named ``sales`` using the configured connector.
 Type mapping
 ------------
 
+The data type mappings are as follows:
+
+=================== ================================ =======================================================================
+PostgreSQL          Trino                            Notes
+=================== ================================ =======================================================================
+``BIT``             ``BOOLEAN``
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/postgresql.rst:101]: PostgreSQL has a real boolean type too.

Also do we properly handle `bit(n)` in the connector?

File: docs/src/main/sphinx/connector/postgresql.rst [hunk 3]
```diff
@@ -93,6 +93,34 @@ catalog named ``sales`` using the configured connector.
...
+``SMALLINT``        ``SMALLINT``
+``INTEGER``         ``INTEGER``
+``BIGINT``          ``BIGINT``
+``REAL``            ``DATE``
+``DOUBLE``          ``DOUBLE``
+``NUMERIC(p, s)``   ``DECIMAL(p, s)``                See :ref:`postgresql-decimal-type-handling` for more information.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/postgresql.rst:107]: Do mention `DECIMAL(p, s)` is an alias of `NUMERIC(p, s)`.

File: docs/src/main/sphinx/connector/postgresql.rst [hunk 4]
```diff
@@ -93,6 +93,34 @@ catalog named ``sales`` using the configured connector.
...
+``CHAR(n)``         ``CHAR(n)``
+``VARCHAR(n)``      ``VARCHAR(n)``
+``BINARY``          ``VARBINARY``
+``DATE``            ``DATE``
+``TIME``            ``TIME``
+``TIMESTAMP``       ``TIMESTAMP``
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/postgresql.rst:113]: we support values with precision too. like below `TIMESTAMPTZ(n)`.

File: docs/src/main/sphinx/connector/postgresql.rst [hunk 5]
```diff
@@ -93,6 +93,34 @@ catalog named ``sales`` using the configured connector.
...
+``TIMESTAMPTZ(n)``  ``TIMESTAMP(n) WITH TIME ZONE``
+``MONEY``           ``VARCHAR``
+``UUID``            ``UUID``
+``JSONB``           ``JSON``                         ``JSONB`` is an alias of ``JSON``.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/postgresql.rst:117]: JSONB and JSON are different types in PostgreSQL. We do handle both though by mapping to Trino JSON.

**Review by hashhar — COMMENTED** [review]
Some of these comments are about adding more test coverage. I'll create separate issues for those.

File: docs/src/main/sphinx/connector/postgresql.rst (hunk 2 above)
Comment (ebyhr) [reply docs/src/main/sphinx/connector/postgresql.rst:101]: > Also do we properly handle bit(n) in the connector?

I don't think so.

Comment (hashhar) [reply docs/src/main/sphinx/connector/postgresql.rst:101]: https://github.com/trinodb/trino/issues/11384

File: docs/src/main/sphinx/connector/postgresql.rst (hunk 1 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/postgresql.rst:105]: https://github.com/trinodb/trino/pull/11385

**Comment by hashhar** [comment]
ENUM is missing. We map that to a VARCHAR. There are tests for it.

File: docs/src/main/sphinx/connector/postgresql.rst [hunk 6]
```diff
@@ -93,6 +93,37 @@ catalog named ``sales`` using the configured connector.
...
+``BIGINT``          ``BIGINT``
+``REAL``            ``DATE``
+``DOUBLE``          ``DOUBLE``
+``NUMERIC(p, s)``   ``DECIMAL(p, s)``                ``DECIMAL(p, s)`` is an alias of  ``NUMERIC(p, s)``.
+                                                     See :ref:`postgresql-decimal-type-handling` for more information.
+``CHAR(n)``         ``CHAR(n)``
+``VARCHAR(n)``      ``VARCHAR(n)``
+``BINARY``          ``VARBINARY``
+``DATE``            ``DATE``
+``TIME``            ``TIME``
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/postgresql.rst:114]: `TIME(n)`

**Review by hashhar — APPROVED** [review]
LGTM % enum + time(n)

**Pull request merged**